### PR TITLE
Fixes from Juelich jureca deployment

### PR DESCRIPTION
**Pull request by pramodk**
- flex & bison dependency missing after mod2c was made submodule

> @jorblancoa : py-bluepy and related python packages are failing to build. (will discuss offline).

### CODE REVIEW

**Comment by pramodk** [comment]
ping @jorblancoa

**Review by jorblancoa — APPROVED** [review]

**Review by jorblancoa — DISMISSED** [review]

File: sysconfig/jureca-booster/modules.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ modules:
           '+profile': 'profile'
           '^coreneuron+knl': 'knl'
           'neuron~mpi': 'serial'
+          '^python@3.6.8': 'python3'
```
Comment (matz-e) [inline sysconfig/jureca-booster/modules.yaml]: Why?

**Review by matz-e — DISMISSED** [review]

File: sysconfig/jureca-booster/modules.yaml (hunk 1 above)
Comment (jorblancoa) [reply sysconfig/jureca-booster/modules.yaml]: We need it for
https://github.com/cnr-ibf-pa/hbp-bsp-issues/issues/525#issuecomment-590703254

**Comment by pramodk** [comment]
Please retest

**Review by matz-e — DISMISSED** [review]

**Comment by pramodk** [comment]
Please retest

File: sysconfig/daint/deploy_simstack.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#spack install --keep-stage neurodamus-mousify+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+
+#spack spec -I py-bluepy%gcc ^python@2.7.15
+#spack install --dirty --keep-stage -v py-bluepy%gcc ^python@2.7.15
+#spack spec -I -l py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15 ^py-tornado@4.4.0 ^py-ipykernel@4.5.0 ^py-ipython@5.1.0
+#spack install --dirty --keep-stage -v py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15
+spack spec -I neuron %intel ^python@2.7.15 ^mpich
+spack install --dirty --keep-stage -v neuron %intel ^python@2.7.15 ^mpich
+
+spack spec -I neuron~mpi %intel ^python@2.7.15
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]:
```suggestion
```

File: sysconfig/daint/deploy_simstack.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# Clone spack repository
+cd $DEPLOYMENT_HOME/sources
+[[ -d spack ]] || git clone https://github.com/BlueBrain/spack.git -b pr/juelich-update
+
+# Setup environment
+export SPACK_ROOT=`pwd`/spack
+export PATH=$SPACK_ROOT/bin:$PATH
+source $SPACK_ROOT/share/spack/setup-env.sh
+
+#chmod -R ugo+w $DEPLOYMENT_HOME
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]:
```suggestion
```

File: sysconfig/daint/deploy_simstack.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+# PYTHON 2 packages
+#spack spec -Il neurodamus-hippocampus+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-hippocampus+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-neocortex+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-mousify+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+
+#spack spec -I py-bluepy%gcc ^python@2.7.15
+#spack install --dirty --keep-stage -v py-bluepy%gcc ^python@2.7.15
+#spack spec -I -l py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15 ^py-tornado@4.4.0 ^py-ipykernel@4.5.0 ^py-ipython@5.1.0
+#spack install --dirty --keep-stage -v py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]:
```suggestion
```

File: sysconfig/daint/deploy_simstack.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+spack install --dirty --keep-stage neurodamus-hippocampus+coreneuron %intel ^python@3.6.5 ^synapsetool%gcc
+spack install --dirty --keep-stage neurodamus-neocortex+coreneuron %intel ^python@3.6.5 ^synapsetool%gcc
+spack install --dirty --keep-stage neurodamus-mousify+coreneuron %intel ^python@3.6.5 ^synapsetool%gcc
+
+spack spec -I neuron~mpi %intel ^python@3.6.5
+spack install --dirty --keep-stage -v neuron~mpi %intel ^python@3.6.5
+
+module swap PrgEnv-intel PrgEnv-gnu
+
+spack spec -I py-bluepy%gcc ^python@3.6.5
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]:
```suggestion
spack spec -Il py-bluepy%gcc ^python@3.6.5
```

File: sysconfig/daint/deploy_simstack.sh [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+spack spec -Il py-sonata-network-reduction%gcc ^python@3.6.5 ^zeromq%intel
+spack install --dirty --keep-stage -v py-sonata-network-reduction%gcc ^python@3.6.5 ^zeromq%intel
+
+spack spec -Il py-bluepyopt%gcc ^python@3.6.5 ^zeromq%intel
+spack install --dirty --keep-stage py-bluepyopt%gcc ^python@3.6.5 ^zeromq%intel
+
+# Re-generate modules
+spack module tcl refresh --delete-tree -y
+cd $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl7-haswell
+find py* -type f -print0|xargs -0 sed -i '/PYTHONPATH.*\/neuron-/d'
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh:69]: add comment here explaining why we do this.

File: sysconfig/daint/deploy_simstack.sh [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+find neuro* -type f -print0|xargs -0 sed -i '/module load mpich/d'
+
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
+#chmod -R ugo-w $DEPLOYMENT_HOME
+
+#cd $BASE_DIR
+#rm modules
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell modules
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]: I think can be removed? I am thinking of not making default symlinks anymore

File: sysconfig/daint/deploy_simstack.sh [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+spack install --dirty --keep-stage py-bluepyopt%gcc ^python@3.6.5 ^zeromq%intel
+
+# Re-generate modules
+spack module tcl refresh --delete-tree -y
+cd $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl7-haswell
+find py* -type f -print0|xargs -0 sed -i '/PYTHONPATH.*\/neuron-/d'
+find neuro* -type f -print0|xargs -0 sed -i '/module load mpich/d'
+
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
+#chmod -R ugo-w $DEPLOYMENT_HOME
```
Comment (pramodk) [inline sysconfig/daint/deploy_simstack.sh]:
```suggestion
ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
chmod -R ugo-w $DEPLOYMENT_HOME
```

File: sysconfig/jureca-booster/deploy_simstack.sh [hunk 9]
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# pass the argument of partition
+if [ "$1" == "" ]; then
+    echo "Error : pass --cluster or --booster"
+    exit 1
+fi
+
+module --force purge
+#module use /usr/local/software/jureca/OtherStages
```
Comment (pramodk) [inline sysconfig/jureca-booster/deploy_simstack.sh]: same

File: sysconfig/jureca-booster/deploy_simstack.sh [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+spack install --keep-stage --dirty py-bluepyopt%gcc ^python@3.6.8 ^zeromq%intel
+
+spack module tcl refresh --delete-tree -y
+cd $DEPLOYMENT_HOME/modules/tcl/linux-centos7-haswell
+find py* -type f -print0|xargs -0 sed -i '/PYTHONPATH.*\/neuron-/d'
+
+# Create symbolic link
+#mkdir -p $DEPLOYMENT_HOME/../install && cd $DEPLOYMENT_HOME/../install
+#rm -f latest
+#ln -s $date/install/modules/tcl/linux-centos7-x86_64 latest
```
Comment (pramodk) [inline sysconfig/jureca-booster/deploy_simstack.sh]: same

File: sysconfig/jureca-booster/deploy_simstack.sh [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+spack install --dirty --keep-stage -v neuron~mpi %intel ^python@3.6.8
+
+spack spec -I py-bluepy%gcc ^python@3.6.8
+spack install --dirty --keep-stage -v py-bluepy%gcc ^python@3.6.8
+
+spack spec -Il py-sonata-network-reduction%gcc ^python@3.6.8 ^zeromq%intel
+spack install --dirty --keep-stage -v py-sonata-network-reduction%gcc ^python@3.6.8 ^zeromq%intel
+
+spack spec -Il py-bluepyopt%gcc ^python@3.6.8 ^zeromq%intel
+spack install --keep-stage --dirty py-bluepyopt%gcc ^python@3.6.8 ^zeromq%intel
```
Comment (pramodk) [inline sysconfig/jureca-booster/deploy_simstack.sh]: nitpick : can we script this bit cleaner? simple changes like: 
```
neurodamus_deps="^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc"
for nd in neurodamus-hippocampus ...
do
   spack install --keep-stage --dirty $nd %intel $neurodamus_deps
done
```

File: sysconfig/daint/deploy_simstack.sh (hunk 7 above)
Comment (jorblancoa) [reply sysconfig/daint/deploy_simstack.sh]: I agree, removed!

File: sysconfig/jureca-booster/deploy_simstack.sh (hunk 10 above)
Comment (jorblancoa) [reply sysconfig/jureca-booster/deploy_simstack.sh]: do you mean to remove everything?

**Review by jorblancoa — APPROVED** [review]

**Pull request merged**